library/systemd/src/lib/yast2/systemd/service.rb

```diff
@@ -0,0 +1,216 @@
+          return [] unless snames.size == property_texts.size
+
+          snames.zip(property_texts).each_with_object([]) do |(name, property_text), memo|
+            service = new(name, propmap, property_text)
+            memo << service unless service.not_found?
+          end
+        end
+
+        # @param service_names [Array<String>] "foo" or "foo.service"
+        # @param propmap [Systemd::Unit::PropMap]
+        # @return [Array<Service,nil>] `nil` if not found
+        def find_many(service_names, propmap = {})
+          services = find_many_at_once(service_names, propmap)
+          return services unless services.empty?
+
+          log.info "Retrying one by one"
+          service_names.map { |n| find(n, propmap) }
+        end
+
+        # @param propmap [Systemd::Unit::PropMap]
+        # @return [Array<Service>]
+        def all(propmap = {})
+          Systemctl.service_units.map { |service_unit| new(service_unit, propmap) }
+        end
+
+        # Instantiate a Systemd::Service object based on the given name
+        #
+        # Use with caution as the service might exist or not. If you need to
+        # react when the service does not exist, use Systemd::Service.find.
+        #
+        # @param service_name [String] "foo" or "foo.service"
+        # @param propmap [Systemd::Unit::PropMap]
+        # @return [Service] System service with the given name
+        def build(service_name, propmap = {})
+          service_name += UNIT_SUFFIX unless service_name.end_with?(UNIT_SUFFIX)
+          new(service_name, propmap)
+        end
+      end
+
+      private_class_method :find_many_at_once
+
+      # Available only on installation system
+      START_SERVICE_INSTSYS_COMMAND = "/bin/service_start".freeze
+
+      # @return [String]
+      def pid
+        properties.pid
+      end
+
+      def running?
+        properties.running?
+      end
+
+      def static?
+        properties.static?
+      end
+
+      def start
+        command = "#{START_SERVICE_INSTSYS_COMMAND} #{unit_name}"
+        installation_system? ? run_instsys_command(command) : super
+      end
+
+      def stop
+        command = "#{START_SERVICE_INSTSYS_COMMAND} --stop #{unit_name}"
+        installation_system? ? run_instsys_command(command) : super
+      end
+
+      def restart
+        # Delegate to Systemd::Unit#restart if not within installation
+        return super unless installation_system?
+
+        stop
+        sleep(1)
+        start
+      end
+
+      # Returns socket associated with service or nil if there is no such socket
+      #
+      # @return [Yast2::Systemd::Socket,nil]
+      # @see Yast2::Systemd::Socket.for_service
+      def socket
+        @socket ||= Yast2::Systemd::Socket.for_service(name)
+      end
+
+      # Determines whether the service has an associated socket
+      #
+      # @return [Boolean] true if an associated socket exists; false otherwise.
+      def socket?
+        !socket.nil?
+      end
+
```

Comment:
`#installation_system?` and `#run_instsys_command` used to be private. Any reason to change that? (I am not against, only curiosity).

See https://github.com/yast/yast-yast2/blob/master/library/systemd/src/modules/systemd_service.rb#L202

Reply:
No reason at all. I guess I deleted the `private` key word accidentally :innocent: The good new it that I am going to restore it ;)

Reply:
BTW, Thank you :)